process_ocaml/408/ProcessExtra.re

```diff
@@ -0,0 +1,616 @@
+  let addForRecord = (recordType, items) => {
+    switch (Shared.dig(recordType).desc) {
+      | Tconstr(path, _args, _memo) => {
+        let t = getTypeAtPath(Shared.mapOldPath(path));
+        items |> List.iter((({Asttypes.txt, loc}, {Types.lbl_res}, _)) => {
+          /* let name = Longident.last(txt); */
+
+          let (name, typeLident) = handleConstructor(path, txt);
+          maybeAddUse(path, typeLident, loc, Attribute(name));
+
+          let nameLoc = Utils.endOfLocation(loc, String.length(name));
+          let locType = switch (t) {
+            | `Local({stamp, contents: {kind: Record(attributes)}}) => {
+              {
+                let%opt_wrap {stamp: astamp} = Belt.List.getBy(attributes, a => a.name.txt == name);
+                addReference(astamp, nameLoc);
+                Loc.LocalReference(stamp, Attribute(name));
+              } |? Loc.NotFound
+            }
+            | `Global(moduleName, path) =>
+              addExternalReference(moduleName, path, Attribute(name), nameLoc);
+              Loc.GlobalReference(moduleName, path, Attribute(name))
+            | _ => Loc.NotFound
+          };
+          addLocation(nameLoc, Loc.Typed(Shared.makeFlexible(lbl_res), locType))
+        })
+      }
+      | _ => ()
+    }
+  };
+
+  let addForConstructor = (constructorType, {Asttypes.txt, loc}, {Types.cstr_name}) => {
+    switch (Shared.dig(constructorType).desc) {
+      | Tconstr(path, _args, _memo) => {
+        /* let name = Longident.last(txt); */
+
+        let (name, typeLident) = handleConstructor(path, txt);
+        maybeAddUse(path, typeLident, loc, Constructor(name));
+
+        let nameLoc = Utils.endOfLocation(loc, String.length(name));
+        let t = getTypeAtPath(Shared.mapOldPath(path));
+        let locType = switch (t) {
+          | `Local({stamp, contents: {kind: Variant(constructos)}}) => {
+            {
+              let%opt_wrap {stamp: cstamp} = Belt.List.getBy(constructos, a => a.name.txt == cstr_name);
+              addReference(cstamp, nameLoc);
+              Loc.LocalReference(stamp, Constructor(name))
+            } |? Loc.NotFound
+          }
+          | `Global(moduleName, path) =>
+            addExternalReference(moduleName, path, Constructor(name), nameLoc);
+            Loc.GlobalReference(moduleName, path, Constructor(name))
+          | _ => Loc.NotFound
+        };
+        addLocation(nameLoc, Loc.Typed(Shared.makeFlexible(constructorType), locType));
+      }
+      | _ => ()
+    }
+  };
+
+  let currentScopeExtent = () => {
+    if (Collector.scopeExtent^ == []) {
+      Location.none
+    } else {
+      List.hd(Collector.scopeExtent^);
+    }
+  };
+  let addScopeExtent = loc => Collector.scopeExtent := [loc, ...Collector.scopeExtent^];
+  let popScopeExtent = () => if (List.length(Collector.scopeExtent^) > 1) {
+    Collector.scopeExtent := List.tl(Collector.scopeExtent^);
+  };
+
+  let rec addForLongident = (top, path: Path.t, txt: Longident.t, loc) => {
```

Comment:
@anmonteiro There were some recent changes merged in `master` that were added to this file: https://github.com/jaredly/reason-language-server/pull/303/files#diff-7d9b7cbbb32d6528ce43ccd07615acfeR261.

Please let me know if you want me to add them, it should be just a copy-paste matter :)

Reply:
Thanks @jchavarri. I'll look at that when I get around to fix the merge conflicts.